### PR TITLE
fix: Add ids to docs headings

### PR DESCRIPTION
**Pull request by mannycarrera4**
## Summary

Adding `id` attribute to `Heading` in the doc parser allows for the doc site to pick up the id so that we can link to specific places in the documentation. This also adds a link icon within storybook!

## Release Category
Documentation

---

### CODE REVIEW

**Review by alanbsmith — APPROVED** [review]
Looks good. Thanks, Manny.

File: modules/docs/lib/widgets/enhancedComponent.tsx [hunk 1]
```diff
@@ -122,7 +129,9 @@ registerWidget<EnhancedComponentValue>('enhancedComponent', ({value, doc, meta})
         ? value.subComponents.map((c, i) => {
             return (
               <React.Fragment key={i}>
-                <Heading>
+                <Heading
+                  id={`${value.displayName?.toLocaleLowerCase()}-${c.name.toLocaleLowerCase}`}
```
Comment (jamesfan) [inline modules/docs/lib/widgets/enhancedComponent.tsx:133]: Something's off with the ids assigned to subcomponents:
Screenshot 2023-07-12 at 12 55 20 PM

Comment (mannycarrera4) [reply modules/docs/lib/widgets/enhancedComponent.tsx:133]: yep!

File: modules/docs/lib/widgets/enhancedComponent.tsx [hunk 2]
```diff
@@ -68,7 +68,17 @@ registerWidget<EnhancedComponentValue>('enhancedComponent', ({value, doc, meta})
     <>
       {value.styleComponent ? (
         <>
-          <Heading headingOffset={1}>Layout Component</Heading>
+          {console.log(parentComponentName)}
+          <Heading
```
Comment (jamesfan) [inline modules/docs/lib/widgets/enhancedComponent.tsx:72]: Got an extra `console.log` here.
```suggestion
          <Heading
```

**Review by jamesfan — APPROVED** [review]
Tested locally with the site, looks good 👍

**Pull request merged**